tests/test_mpathcount.py

```diff
@@ -0,0 +1,203 @@
+"""
+Unit tests for mpathcount
+"""
+import errno
+import os
+import unittest
+import mock
+import testlib
+import mpathcount
+import SR
+import util
+
+#class FakeXapi(object):
+    #def __init__(self):
+        #self.srRecord = {
+            #'name_label': 'dummy'
+        #}
+
+    #def isPluggedHere(self):
+        #return True
+
+    #def isMaster(self):
+        #return True
+
+# pylint: disable=W0613; mocks don't need to be accessed
+# pylint: disable=R0201; methods must be instance for nose to work
+# pylint: disable=W0212; unit tests are permitted to snoop
+class TestMpathCount(unittest.TestCase):
+    """
+    Unit tests for mpathcount
+    """
+
+    @mock.patch('mpathcount.os', autospec=True)
+    def test_get_root_dev_major(self, mock_os):
+        mpathcount.get_root_dev_major()
+        assert(mock_os.major.called)
+
+    @mock.patch('mpathcount.mpath_cli', autospec=True)
+    def test_get_path_count(self, mpath_cli):
+        mpath_cli.get_topology.return_value = [
+            "multipathd> show map 3600a098038303973743f486833396d44 topology",
+            "3600a098038303973743f486833396d44 dm-1 NETAPP  ,LUN C-Mode",
+            "size=200G features='4 queue_if_no_path pg_init_retries 50 retain_attached_hw_handle' hwhandler='1 alua' wp=rw",
+            "`-+- policy='service-time 0' prio=50 status=active",
+            "  |- 0:0:0:4  sdr  65:16  active ready  running",
+            "  |- 0:0:1:4  sdg  8:96   active ready  running",
+            "  |- 7:0:0:4  sdab 65:176 failed faulty running",
+            "  `- 7:0:1:4  sdam 66:96  failed faulty running"
+            ]
+        count, total = mpathcount.get_path_count('3600a098038303973743f486833396d44')
+        self.assertEqual(4, total, msg='total should be 4, was {}'.format(total))
```

Comment:
the format is unnecessary, the unit test assert does all that anyway. Something like "Total count incorrect" is sufficient